### ISSUE TITLE
Add the user to the databse

### ISSUE THREAD

**Issue by hrishikeshpaul** [hrishikeshpaul/sailboat#29]
After the user logs in,

1. If they are logging in for the first time, create a new object and add them. This should happen in the `/authorize` route. No need to return the user.
2. If they are already an user don't do anything.
---

**Side Note**:

The motivation is that once the user passes the login page and is authenticated by google/facebook, the frontend will make a `/me` call to get the user details. This is beneficial because
1. In case of page refreshes. 
2. If the URL hits a private route direct, and is in session.
3. 
Basically ensuring that user is persists once they have logged in.
